### PR TITLE
Add nonce JSON-RPC endpoint

### LINKED ISSUE #63: Feature request: Add RPC endpoint for retrieving the nonce from the DIDRegistry smart contract
### Is your feature request related to a problem?

In order to sign things we need the nonce, but xps-gateway does not provide a way to retrieve this so libxmtp ends up pulling in DIDRegistry. in order to remove the dependency on DIDRegistry, xps-gateway should include an endpoint for retrieving the nonce.

### Describe the solution to the problem

the DIDRegistry includes a smart-contract call, `nonce` that should be incorporated into a JSON-RPC endpoint

### PR DESCRIPTION
**Pull request by insipx**
Adds a nonce endpoint in order to sign setAttribute payloads in `libxmtp` and `xps-cli`

- adds documentation for `walletAddress`
- fixes docs in some places
- adds `nonce` endpoint & tests
- tests for `resolve_did_address`

closes #63

### CODE REVIEW

**Review by 37ng — APPROVED** [review]
lgtm
not in the scope of this pr - I think our doc could be more concise and omit information that could be found in [JSON-RPC 2.0 Specification ](https://www.jsonrpc.org/specification)

File: lib-xps/src/rpc/api.rs [hunk 1]
```diff
@@ -517,14 +517,90 @@ pub trait Xps {
     #[method(name = "status")]
     async fn status(&self) -> Result<String, ErrorObjectOwned>;
 
+    /// ### Documentation for JSON RPC Endpoint: `xps_walletAddress`
```
Comment (jac18281828) [inline lib-xps/src/rpc/api.rs:520]: 🔥

File: lib-xps/src/rpc/api.rs [hunk 2]
```diff
@@ -601,4 +677,88 @@ pub trait Xps {
...
+    ///     - `jsonrpc`: `"2.0"`
+    ///     - `method`: `"xps_nonce"`
+    ///     - `params`: Array (optional parameters as required)
+    ///     - `id`: Request identifier (integer or string)
+    /// **Example Request Body:**
+    /// ```json
+    /// {
+    /// "jsonrpc": "2.0",
+    /// "method": "xps_nonce",
+    /// "params": ["0x0000000000000000000000000000000000000000"],
```
Comment (jac18281828) [inline lib-xps/src/rpc/api.rs:704]: I recommend a random address for the docs vs 0, in ethereum 0 has a special meaning (deployer address)

**Review by jac18281828 — APPROVED** [review]
All looks good!  Thanks for adding this

**Pull request merged**